lib/internal/modules/cjs/helpers.js

```diff
@@ -32,11 +32,14 @@ let debug = require('internal/util/debuglog').debuglog('module', (fn) => {
 // TODO: Use this set when resolving pkg#exports conditions in loader.js.
 const cjsConditions = new SafeSet(['require', 'node', ...userConditions]);
 
-function loadNativeModule(filename, request) {
+function loadNativeModule(filename, request, skipCompile = false) {
   const mod = NativeModule.map.get(filename);
   if (mod) {
     debug('load native module %s', request);
-    mod.compileForPublicLoader();
+    if (skipCompile) {
+      // compileForPublicLoader() throws if mod.canBeRequiredByUsers is false:
```

Comment:
The error from compileForPublicLoader is not documented as a public error and does not have an error code - https://github.com/nodejs/node/blob/master/lib/internal/bootstrap/loaders.js#L223 (and see the associated comment). I think this error should probably be upgraded into a coded error if exposed this way.

Reply:
It’s meant to throw an `ERR_INTERNAL_ASSERTION`, but the `errors.js` module doesn’t exist yet at that point.

Reply:
Right, in which case, perhaps it would be better to just return `undefined` and rewrap the error on the other side via a helper function that does have access to errors?